components/builder-api/src/server/mod.rs

```diff
@@ -143,8 +152,10 @@ pub fn run(config: Config) -> Result<()> {
         cfg.listen_port()
     );
 
+    let addr = init(config.datastore.clone());
+
     server::new(move || {
-        let app_state = AppState::new(&config);
+        let app_state = AppState::new(&config, addr.clone());
```

Comment:
Is this creating N different datastore pools?

Reply:
No, just a single pool inside the main actix reactor